plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1750,9 +1746,13 @@ void producer_plugin_impl::schedule_production_loop() {
          auto expect_time = chain.pending_block_time() - fc::microseconds(config::block_interval_us);
          // ship this block off up to 1 block time earlier or immediately
          if (fc::time_point::now() >= expect_time) {
-            _timer.expires_from_now( boost::posix_time::microseconds( 0 ));
-            fc_dlog(_log, "Scheduling Block Production on Exhausted Block #${num} immediately",
+            // produce block immediately
```

Comment:
While I cannot find a direct reason this is an issue, there are a few implicit requirements that this may one day run-afoul of that I think we should attempt to make more explicit

1. `schedule_production_loop` is almost always called from a destructor of an `fc::scoped_exit` which means it really shouldn't throw (maybe one day we should consider requiring `noexcept` qualifiers on lambdas for `fc::scoped_exit).  Right now, `maybe_produce_block` appears to handle all of the exceptions and is clean but I'd be interested in how we make sure the implicit requirement that it not throw is made explicit. 

2. `maybe_produce_block` calls `schedule_production_loop` making this recursive.  I don't think it can get into an infinite recursion because this branch is taken based on time but, there are a lot of variables and if this stays direct, perhaps we should "schedule" the call to `schedule_production_loop` in `maybe_produce_block` so that it at least clears the stack ?

at the least we should probably make the log messages the same (other than the  notice about exhaustion) and turn the call and the `fc_dlog`calls into a local lambda so that the messages are kept consistent.

Reply:
Reverted this change as I think it is clearer and more consistent the way it was.